pandas/tests/frame/test_reductions.py

```diff
@@ -1145,29 +1145,43 @@ def test_any_all_object(self):
         result = np.any(DataFrame(columns=["a", "b"])).item()
         assert result is False
 
-    def test_any_all_object_bool_only(self):
+    def test_any_all_object_bool_only(self, using_array_manager):
         df = DataFrame({"A": ["foo", 2], "B": [True, False]}).astype(object)
         df._consolidate_inplace()
         df["C"] = Series([True, True])
 
         # The underlying bug is in DataFrame._get_bool_data, so we check
         #  that while we're here
         res = df._get_bool_data()
-        expected = df[["B", "C"]]
+
+        # With ArrayManager currently don't infer object dtype to be boolean
```

Comment:
i would like to deprecate this behavior in ObjectBlock.is_bool, but until then, can we keep the behaviors in sync?

Reply:
Yeah, was already planning to add it back (but first need to actually implement get_bool_data to honor that). Will update tomorrow